packages/cli/src/swagger/specGenerator3.ts

```diff
@@ -280,6 +281,28 @@ export class SpecGenerator3 extends SpecGenerator {
       throw new Error('Either body parameter or form parameters allowed per controller method - not both.');
     }
 
+    if (bodyPropParams.length > 1) {
+      if (!bodyParams.length) {
+        bodyParams.push({
+          in: 'body',
+          name: 'body',
+          parameterName: 'body',
+          required: true,
+          type: {
+            dataType: 'nestedObjectLiteral',
+            properties: [],
+          } as Tsoa.NestedObjectLiteralType,
+          validators: {},
+          deprecated: false,
+        });
+      }
+
+      const type: Tsoa.NestedObjectLiteralType = bodyParams[0].type as Tsoa.NestedObjectLiteralType;
+      bodyPropParams.forEach((bodyParam: Tsoa.Parameter) => {
+        type.properties.push(bodyParam as Tsoa.Property);
+      });
```

Comment:
This should break in scenarios where we have a ref type as a body.

Reply:
Body and BodyProps couldn't be in same place. It will through exception. You have tests, just launch them....

Reply:
The tests are executed automatically in CI, but thanks for the tip :)
But you are somewhat correct, we don't have any tests that have controller methods expecting both `Body` and `BodyProp`. Bit of an undocumented caveat, but I don't think we can expect that within this PR.